packages/flutter/lib/src/material/tabs.dart

```diff
@@ -381,19 +381,61 @@ class _TabsScrollBehavior extends BoundedBehavior {
   }
 }
 
-class TabBarSelection {
-  TabBarSelection({ int index: 0, this.maxIndex, this.onChanged }) : _index = index {
+class _TabBarSelection extends InheritedWidget {
+  _TabBarSelection({
+    this.selection,
+    Key key,
+    Widget child
+  }) : super(key: key, child: child);
+
+  final TabBarSelectionState selection;
+
+  bool updateShouldNotify(_TabBarSelection oldWidget) => selection != oldWidget.selection;
```

Comment:
I suspect the reason it works is that you're hooking into the tab bar selection's performance, and that's what's driving the changes, rather than the inheritedness.